src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -157,73 +127,35 @@ export class CdkMenuBar extends CdkMenuGroup implements Menu, AfterContentInit,
     }
   }
 
-  /** Setup the FocusKeyManager with the correct orientation for the menu bar. */
-  private _setKeyManager() {
-    this._keyManager = new FocusKeyManager(this._allItems)
-      .withWrap()
-      .withTypeAhead()
-      .withHomeAndEnd();
-
-    if (this._isHorizontal()) {
-      this._keyManager.withHorizontalOrientation(this._dir?.value || 'ltr');
-    } else {
-      this._keyManager.withVerticalOrientation();
-    }
-  }
-
   /**
    * Set the PointerFocusTracker and ensure that when mouse focus changes the key manager is updated
    * with the latest menu item under mouse focus.
    */
   private _subscribeToMouseManager() {
     this._ngZone.runOutsideAngular(() => {
-      this._pointerTracker = new PointerFocusTracker(this._allItems);
-      this._pointerTracker.entered.pipe(takeUntil(this._destroyed)).subscribe(item => {
-        if (this._hasOpenSubmenu()) {
-          this._keyManager.setActiveItem(item);
+      this.pointerTracker = new PointerFocusTracker(this.items);
+      this.pointerTracker.entered.pipe(takeUntil(this.destroyed)).subscribe(item => {
+        // TODO(mmalerba): Why does the menu bar do this in a hasOpenSubmenu check
+        //   when menu just does it regardless?
+        if (this.hasOpenSubmenu()) {
+          this.keyManager.setActiveItem(item);
         }
       });
     });
   }
 
-  /** Subscribe to the MenuStack close and empty observables. */
-  private _subscribeToMenuStack() {
-    this._menuStack.closed
-      .pipe(takeUntil(this._destroyed))
-      .subscribe(item => this._closeOpenMenu(item));
-
-    this._menuStack.emptied
-      .pipe(takeUntil(this._destroyed))
-      .subscribe(event => this._toggleOpenMenu(event));
-  }
-
-  /**
-   * Close the open menu if the current active item opened the requested MenuStackItem.
-   * @param item the MenuStackItem requested to be closed.
-   */
-  private _closeOpenMenu(menu: MenuStackItem | undefined) {
-    const trigger = this._openItem;
-    const keyManager = this._keyManager;
-    if (menu === trigger?.getMenuTrigger()?.getMenu()) {
-      trigger?.getMenuTrigger()?.closeMenu();
-      // If the user has moused over a sibling item we want to focus the element under mouse focus
-      // not the trigger which previously opened the now closed menu.
-      if (trigger) {
-        keyManager.setActiveItem(this._pointerTracker?.activeElement || trigger);
-      }
-    }
-  }
-
   /**
    * Set focus to either the current, previous or next item based on the FocusNext event, then
    * open the previous or next item.
    */
   private _toggleOpenMenu(event: FocusNext | undefined) {
-    const keyManager = this._keyManager;
+    const keyManager = this.keyManager;
     switch (event) {
       case FocusNext.nextItem:
         keyManager.setFocusOrigin('keyboard');
         keyManager.setNextItemActive();
+        // TODO(mmalerba): This seems similar to what menu does, but menu bar also calls openMenu
```

Comment:
Calling out this TODO in case anyone has an idea why menu is different from menubar in this way

Reply:
Ok I think I understand why this is different. In drive when you arrow between the "file", "edit", etc. menus in the menu bar it automatically pops them open, but sub-menus are not automatically opened you have to press right arrow / space / enter to expand them